backend/lib/routes/index.js

```diff
@@ -0,0 +1,48 @@
+const apiRoute = require('./home');
+const userRoute = require('./users');
+const tagsRoute = require('./tags');
+const { appendPath, deepmerge } = require('../utils/');
+
+const routes = [
+    apiRoute,
+    userRoute,
+    tagsRoute,
+];
+
+/**
+ * Makes the child object inherit the args of the parent object, but keeps the overwrites
+ * @param {Object} parent parent who's args will be inherited
+ * @param {Object} child child who will inherit the args
+ * @returns {undefined}
+ */
+const inheritArgs = (parent, child) => {
+    child.args = deepmerge(parent.args || {}, child.args || {});
+};
+
+/**
+ * Route binder, used to bind routes to the httpServer
+ * @param {Object} http httpServer that the routes should be bound on
+ * @param {Object} route the route that should be bound
+ * @param {String} [parentPath=''] the base path of the route
+ * @returns {undefined}
+ */
+const bindRoute = (http, route, parentPath = '') => {
+    const localPath = appendPath(parentPath, route.path, route.pathOption);
+
+    if (route.method && route.controller) {
+        http[route.method](localPath, route.controller, route.args);
```

Comment:
@noxi333 should we check on any duplicates?

Reply:
I don't think so, express allows it and if there is a usecase for overriding existing routes, then it should be doable